### PR TITLE
Fix Decimal issue on swap page.

### LINKED ISSUE #1101: [ETH] Decimal issue on swap page.
Go to ETH/RUNE or ETH/BNB swap page.
User can input ETH amount which is larger than the account balance.
Even it's able to drag to swap.

It seems a decimal handling issue.

### PR DESCRIPTION
**Pull request by junkai121**
close #1101 

- make sure the target amount decimal is the same as source asset decimal.

### CODE REVIEW

File: src/renderer/components/swap/Swap.tsx [hunk 1]
```diff
@@ -332,6 +332,9 @@ export const Swap = ({
 
   const setAmountToSwap = useCallback(
     (targetAmount: BaseAmount) => {
+      // make sure the target amount decimal is the same as source asset decimal.
+      targetAmount.decimal = sourceAssetAmount.decimal
```
Comment (veado) [inline src/renderer/components/swap/Swap.tsx:336]: Imagine: 
- source: ETH (18 decimal)
- target: BTC (8 decimal)

With this change ^ BTC will have 18 decimal ...

**Review by veado — CHANGES_REQUESTED** [review]

File: src/renderer/components/swap/Swap.tsx (hunk 1 above)
Comment (junkai121) [reply src/renderer/components/swap/Swap.tsx:336]: Here, the targetAmount is the user input for source asset.

Comment (veado) [reply src/renderer/components/swap/Swap.tsx:336]: Ah, I've missed that ^ ...

**Review by veado — APPROVED** [review]
Good find!

**Pull request merged**